craft_parts/sources/base.py

```diff
@@ -77,7 +77,7 @@ def __init__(
         self.source_branch = source_branch
         self.source_depth = source_depth
         self.source_checksum = source_checksum
-        self.source_details = None
+        self.source_details: Any = None
```

Comment:
```suggestion
        self.source_details: Optional[Dict[str, Optional[str]]] = None
```

Are other types of sources planning to provide something else? (Right now only `git_source` provides details)

Reply:
Updated to `Optional[Dict[str, Optional[str]]]`, if a subclass changes this in the future we can revisit it